pkg/config/config.go

```diff
@@ -208,6 +210,29 @@ func New(opts ...Option) (*Config, error) {
 			absoluteConfigPaths[i] = path
 		}
 	}
+
+	// allow the users to set datadir to a path like ~/.bacalhau or ~/something/idk/whatever
+	// and expand the path for them
+	dataDirPath := c.base.GetString(types.DataDirKey)
+	if dataDirPath[0] == '~' {
+		log.Info().Msgf("configuration field 'DataDir' contains '~': (%s). Attempting to expand to the home directory...", dataDirPath)
+		expanded, err := homedir.Expand(dataDirPath)
+		if err == nil {
+			dataDirPath = expanded
+			c.base.Set(types.DataDirKey, dataDirPath)
+			log.Info().Msgf("successfully expanded data directory to %s", dataDirPath)
+		}
+	}
```

Comment:
I am missing the purpose of expanding `~`. Also this is going to generate a lot of unnecessary logs for users. In most cases when I define a dataDir I use `~` and I believe most of our users do as we keep saying `~/.bacalahu` is the default repo

Reply:
> I am missing the purpose of expanding ~

see description here: https://github.com/bacalhau-project/bacalhau/pull/4485#issuecomment-2364714820

> Also this is going to generate a lot of unnecessary logs for users

Okay, I will remove the logs.

> In most cases when I define a dataDir I use ~ and I believe most of our users do as we keep saying ~/.bacalahu is the default repo

Thats the intention, but we need to expand it to a valid path before using it internally for filesystem operations.